### PR TITLE
integrate with google cloud gcp relation and external-cloud-provider

### PR DESCRIPTION
**Pull request by addyess**
Tests integration with Google Cloud and external provider
* gcp-cloud-provider
* gcp-k8s-storage
* gcp-integrator charms through the ops version of the gcp-integration

Depends on merges of:
* https://github.com/charmed-kubernetes/interface-gcp-integration/pull/1
* https://github.com/charmed-kubernetes/charm-lib-kubernetes-snaps/pull/15

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ charm-lib-interface-external-cloud-provider @ git+https://github.com/charmed-kub
 charm-lib-interface-kube-dns @ git+https://github.com/charmed-kubernetes/charm-lib-interface-kube-dns
 charm-lib-interface-kubernetes-cni @ git+https://github.com/charmed-kubernetes/charm-lib-interface-kubernetes-cni
 charm-lib-interface-tokens @ git+https://github.com/charmed-kubernetes/charm-lib-interface-tokens
-charm-lib-kubernetes-snaps @ git+https://github.com/charmed-kubernetes/charm-lib-kubernetes-snaps
+charm-lib-kubernetes-snaps @ git+https://github.com/charmed-kubernetes/charm-lib-kubernetes-snaps@KU-439/gcp-cloud-integration
```
Comment (kwmonroe) [inline requirements.txt:8]:
```suggestion
charm-lib-kubernetes-snaps @ git+https://github.com/charmed-kubernetes/charm-lib-kubernetes-snaps
```

File: requirements.txt [hunk 2]
```diff
@@ -14,8 +14,10 @@ interface_hacluster @ git+https://github.com/charmed-kubernetes/charm-interface-
 jinja2
 loadbalancer_interface
 ops >= 2.2.0
-ops.interface_kube_control @ git+https://github.com/juju-solutions/interface-kube-control#subdirectory=ops
-ops.interface_tls_certificates @ git+https://github.com/juju-solutions/interface-tls-certificates#subdirectory=ops
-ops.interface_aws @ git+https://github.com/juju-solutions/interface-aws-integration#subdirectory=ops
+ops.interface_kube_control @ git+https://github.com/charmed-kubernetes/interface-kube-control@main#subdirectory=ops
+ops.interface_tls_certificates @ git+https://github.com/charmed-kubernetes/interface-tls-certificates@main#subdirectory=ops
+ops.interface_aws @ git+https://github.com/charmed-kubernetes/interface-aws-integration@main#subdirectory=ops
+ops.interface_gcp @ git+https://github.com/charmed-kubernetes/interface-gcp-integration@KU-439/gcp-ops-integration#subdirectory=ops
```
Comment (kwmonroe) [inline requirements.txt:20]:
```suggestion
ops.interface_gcp @ git+https://github.com/charmed-kubernetes/interface-gcp-integration@main#subdirectory=ops
```

File: src/cloud_integration.py [hunk 3]
```diff
@@ -65,7 +76,7 @@ def integrate(self, event: ops.EventBase):
             cloud.enable_autoscaling_readonly()
             cloud.enable_instance_modification()
             cloud.enable_region_readonly()
-        elif cloud_name == "gcp":
+        elif cloud_name == "gce":
```
Comment (kwmonroe) [inline src/cloud_integration.py:79]: similar comment as to k-w; we're double sure the charm knows this cloud as `gce` and not `gcp`?

**Review by kwmonroe — APPROVED** [review]
suggested an update to the requirements branches given the preerqs are merged; given that, lgtm.

File: src/cloud_integration.py (hunk 3 above)
Comment (addyess) [reply src/cloud_integration.py:79]: yep... `get_cloud_name` is handled by [this library](https://github.com/charmed-kubernetes/charm-lib-interface-external-cloud-provider/blob/main/charms/interface_external_cloud_provider.py#L13)

**Review by kwmonroe — APPROVED** [review]

**Pull request merged**